src/sentry/integrations/opsgenie/integration.py

```diff
@@ -156,10 +164,52 @@ def update_organization_config(self, data: MutableMapping[str, Any]) -> None:
         # this is not instantaneous, so you could add the same team a bunch of times in a row
         # but I don't anticipate this being too much of an issue
         added_names = {team["team"] for team in teams if team not in unsaved_teams}
+        existing_team_key_pairs = {
+            (team["team"], team["integration_key"]) for team in teams if team not in unsaved_teams
+        }
+
+        integration = integration_service.get_integration(
+            organization_integration_id=self.org_integration.id
+        )
+        if not integration:
+            raise IntegrationError("Integration does not exist")
+
         for team in unsaved_teams:
             if team["team"] in added_names:
                 raise ValidationError({"duplicate_name": ["Duplicate team name."]})
             team["id"] = str(self.org_integration.id) + "-" + team["team"]
+
+        for team in teams:
+            # skip if team, key pair already exist in config
+            if (team["team"], team["integration_key"]) in existing_team_key_pairs:
+                continue
+
+            integration_key = team["integration_key"]
+
+            # validate integration keys
+            client = OpsgenieClient(
+                integration=integration,
+                integration_key=integration_key,
+            )
+            # call an API to test the integration key
+            try:
+                client.get_teams()
+            except ApiError as e:
+                logger.info(
+                    "opsgenie.authorization_error",
+                    extra={"error": str(e), "status_code": e.code},
+                )
+                if e.code == 429:
+                    raise ApiRateLimitedError(
+                        "Too many requests. Please try updating one team/key at a time."
+                    )
+                elif e.code == 401:
+                    raise ApiUnauthorized(f"Invalid integration key {integration_key}")
```

Comment:
so it rises only on the first one that's wrong? 
maybe it would make sense to go through all of them and return all of the ones that are wrong?

Reply:
so collect all of the 401s and raise a single error? that does sound better